### ISSUE TITLE
Add ServiceStack Reference provides TypeScript code that does not compile

### ISSUE THREAD

**Issue by MortenChristiansen** [ServiceStack/ServiceStackVS#41]
Generally, the generated dto file has a compilation error because `IReturn<>` does not have a `T` type parameter. The classes implementing the interface raise the error "Type 'IReturn' is not generic". So far I have just added the `T` manually every time I regenerate the file but since turning on the compiler option `noUnusedLocals` I instead get the error "'T' is declared but never used". I suggest that the generated code for `IReturn<>` should be changed to this, which works for me:

```typescript
export interface IReturn<T>
{
    createResponse(): T;
}
```

In case you're experience is different than mine, here is my tsconfig settings:

```json
{
  "compilerOptions": {
    "noImplicitAny": false,
    "noEmitOnError": true,
    "removeComments": false,
    "sourceMap": true,
    "target": "es2015",
    "module": "system",
    "jsx": "react",
    "experimentalDecorators": true,
    "outDir": "../wwwroot/app",
    "noUnusedLocals": true,
    "strictNullChecks": true
  },
  "exclude": [
    "jspm_packages",
    "typings",
    "node_modules",
    "wwwroot"
  ],
  "compileOnSave": true
}

```

Typescript version: 2.1.4.

**Comment by mythz** [comment]
Great timing, [this made it](https://github.com/ServiceStack/ServiceStack/commit/28d6956409a33b9c1e5f593e2860a2f5555dcc28) into the [v4.5.6 that was just released](http://docs.servicestack.net/releases/v4.5.6).

Thanks for the suggestion!

**Comment by MortenChristiansen** [comment]
It still does not work for me.

```
interface IReturn<>
{
    createResponse() : T;
}
```

should be

```
interface IReturn<T>
{
    createResponse() : T;
}
```

**Comment by MortenChristiansen** [comment]
Thinking about it a bit more, `IReturnVoid` should also have `createResponse(): {};`.

**Comment by mythz** [comment]
There is no `IReturn<>`, there's only `IReturn<T>`, what compiler errors are you getting?

Can you provide the generated DTOs which are causing issues?

**Comment by MortenChristiansen** [comment]
I have just generated the following dto file: https://gist.github.com/MortenChristiansen/f09f312ab1423dd606954477fda57510

As you can see, the `T` is missing from the interface.

**Comment by mythz** [comment]
ok that's strange, what's your environment .NET Core Linux/Windows?

**Comment by mythz** [comment]
I was able to repro this [compatibility issue with .NET Core](http://stackoverflow.com/a/39140220/85785) which is now fixed from [this commit](https://github.com/ServiceStack/ServiceStack.Text/commit/11f487f5072b036c27e64ba69415a063fc1e3dcd).

This change is available from .NET Core v1.0.33+ that's now available on NuGet. Running a `dotnet restore` should pull down the latest `v1.0.33` packages from NuGet.
